src/main/scala/is/hail/variant/MatrixTable.scala

```diff
@@ -799,43 +799,6 @@ class MatrixTable(val hc: HailContext, val ast: MatrixIR) {
     copyMT(matrixType = matrixType.copy(colType = t), colValues = colValues.copy(value = newAnnotations, t = TArray(t)))
   }
 
-  def annotateColsExpr(expr: String): MatrixTable = {
-    val ec = colEC
-
-    val (paths, types, f) = Parser.parseAnnotationExprs(expr, ec, None)
-
-    val inserterBuilder = new ArrayBuilder[Inserter]()
-    val finalType = (paths, types).zipped.foldLeft(colType) { case (sas, (ids, signature)) =>
-      val (s, i) = sas.structInsert(signature, ids)
-      inserterBuilder += i
-      s
-    }
-    val inserters = inserterBuilder.result()
-
-    val colAggregationOption = Aggregators.buildColAggregations(hc, value, ec)
-
-    ec.set(0, globals.value)
-
-    val newAnnotations = new Array[Annotation](numCols)
-
-    var i = 0
-    while (i < numCols) {
-      colAggregationOption.foreach(_.apply(i))
-      val sa = colValues.value(i)
-      ec.set(1, sa)
-
-      newAnnotations(i) = f().zip(inserters)
-        .foldLeft(sa) { case (sa, (v, inserter)) =>
-          inserter(sa, v)
-        }
-      i += 1
-    }
-
-    val saFields = finalType.fieldNames.toSet
-    copyMT(colValues = colValues.copy(value = newAnnotations, t = TArray(finalType)),
-      matrixType = matrixType.copy(colKey = colKey.filter(saFields.contains), colType = finalType))
-  }
-
   def annotateCols(annotations: Map[Annotation, Annotation], signature: Type, root: String): MatrixTable = {
```

Comment:
Should this get added to RichMatrixTable? Is it still used anywhere?

Reply:
I did. (slightly different form, but I updated the tests accordingly).

Reply:
Can you delete it from MatrixTable.scala?

Reply:
I mean `annotateCols`

Reply:
oh, this one. my bad I thought you were referring to the other one. I'll move this also

Reply:
(done)